src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -79,6 +82,25 @@ public bool ManualSettingsOnly
 
         public string ProxyBypass { get; set; }
 
+        public bool RecentAutoDetectionFailure
+        {
+            get
+            {
+                if (!_autoDetectionFailed)
+                {
+                    return false;
+                }
+                else if ((DateTime.Now - _lastTimeAutoDetectionFailed) <= _recentAutoDetectionInterval)
```

Comment:
I'd prefer we use Environment.TickCount (which is milliseconds) rather than DateTime.Now (and if we really wanted to use DateTime, it'd be better to use DateTime.UtcNow and avoid arithmetic problems with time zones and daylight savings as well as the extra costs associated with time zone lookup and manipulation).  Environment.TickCount is a bit faster than DateTime.UtcNow and much faster than DateTime.UtcNow, and while it has less precision/accuracy, we also don't need a lot of precision here.

Reply:
I'll switch to Environment.TickCount.